src/components/can_cooperation/src/commands/set_interior_vehicle_data_request.cc

```diff
@@ -0,0 +1,146 @@
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "can_cooperation/commands/set_interior_vehicle_data_request.h"
+#include <algorithm>
+#include "can_cooperation/can_module_constants.h"
+#include "can_cooperation/message_helper.h"
+#include "functional_module/function_ids.h"
+#include "json/json.h"
+
+namespace can_cooperation {
+
+namespace commands {
+
+using namespace json_keys;
+using namespace message_params;
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "SetInteriorVehicleDataRequest")
+
+SetInteriorVehicleDataRequest::SetInteriorVehicleDataRequest(
+    const application_manager::MessagePtr& message,
+    CANModuleInterface& can_module)
+    : BaseCommandRequest(message, can_module) {}
+
+SetInteriorVehicleDataRequest::~SetInteriorVehicleDataRequest() {}
+
+void SetInteriorVehicleDataRequest::Execute() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const Json::Value request_params =
+      MessageHelper::StringToValue(message_->json_message());
+  const Json::Value module_data = request_params[kModuleData];
+  const std::string module_type = module_data[kModuleType].asString();
+  bool module_type_and_data_match = true;
+
+  if (enums_value::kRadio == module_type) {
+    module_type_and_data_match = !IsMember(module_data, kClimateControlData);
+  }
+
+  if (enums_value::kClimate == module_type) {
+    module_type_and_data_match = !IsMember(module_data, kRadioControlData);
+  }
+
+  if (module_type_and_data_match) {
+    SendRequest(functional_modules::hmi_api::set_interior_vehicle_data,
+                request_params,
+                true);
+  } else {
+    LOG4CXX_WARN(logger_, "Request module type & data mismatch!");
+    SendResponse(false, result_codes::kInvalidData, "");
+  }
+}
+
+void SetInteriorVehicleDataRequest::OnEvent(
+    const can_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK_OR_RETURN_VOID(
+      (functional_modules::hmi_api::set_interior_vehicle_data == event.id()));
+
+  application_manager::Message& hmi_response = *(event.event_message());
+  const bool validate_result = application_manager::SUCCESS ==
+                               service()->ValidateMessageBySchema(hmi_response);
+  LOG4CXX_DEBUG(logger_,
+                "HMI response validation result is " << validate_result);
+  const Json::Value value =
+      MessageHelper::StringToValue(hmi_response.json_message());
+
+  std::string result_code;
+  std::string info;
+
+  bool success = false;
+  if (IsMember(value, kError) && IsMember(value[kError], kCode) &&
```

Comment:
@AByzhynar as I could see this part of logic is implemented in `BaseCommandRequest::ParseResultCode`
Why did you move it here?